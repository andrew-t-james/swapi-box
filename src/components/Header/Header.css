.header {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: 0 2rem;
  z-index: 1;
}

.header__title {
  font-size: 3rem;
  font-weight: 100;
  letter-spacing: 1px;
}

.header__nav {
  list-style: none;
}