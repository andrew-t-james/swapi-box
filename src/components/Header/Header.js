import React from 'react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

import './Header.css';

const Header = ({ updateSelection, selected }) => {
  return (
    <header className="header">
      <h2 className="header__title">SWapiBox</h2>
      <ul className="header__nav">
        <li className={`${selected === 'favorites' ? 'btn-active' : ''}`}
          onClick={() => updateSelection()}>
          <Button
            customStyle={'card-container__nav--button btn'}
            name="Favorites"
          />
        </li>
      </ul>
    </header>
  );
};

Header.propTypes = {
  updateSelection: PropTypes.func,
  selected: PropTypes.string
};

export default Header;
