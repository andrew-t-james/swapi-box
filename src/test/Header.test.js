import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header/Header';

describe('Header', () => {
  let wrapper;
  const mockFunc = jest.fn();

  beforeEach(() => wrapper = shallow(<Header updateSelection={mockFunc} favorites={[{}, {}]}/>));

  afterEach(() => wrapper.unmount());

  test('should call selectFavorites on click', () => {
    wrapper.find('li').simulate('click');
    expect(mockFunc).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
