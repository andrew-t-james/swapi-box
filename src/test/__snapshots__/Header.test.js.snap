// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    updateSelection={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="header__title"
>
          SWapiBox
</h2>,
        <ul
          className="header__nav"
>
          <li
                    className=""
                    onClick={[Function]}
          >
                    <Button
                              customStyle="card-container__nav--button btn"
                              name="Favorites"
                    />
          </li>
</ul>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWapiBox",
          "className": "header__title",
        },
        "ref": null,
        "rendered": "SWapiBox",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li
            className=""
            onClick={[Function]}
>
            <Button
                        customStyle="card-container__nav--button btn"
                        name="Favorites"
            />
</li>,
          "className": "header__nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              customStyle="card-container__nav--button btn"
              name="Favorites"
/>,
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "customStyle": "card-container__nav--button btn",
              "name": "Favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="header__title"
>
            SWapiBox
</h2>,
          <ul
            className="header__nav"
>
            <li
                        className=""
                        onClick={[Function]}
            >
                        <Button
                                    customStyle="card-container__nav--button btn"
                                    name="Favorites"
                        />
            </li>
</ul>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWapiBox",
            "className": "header__title",
          },
          "ref": null,
          "rendered": "SWapiBox",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li
              className=""
              onClick={[Function]}
>
              <Button
                            customStyle="card-container__nav--button btn"
                            name="Favorites"
              />
</li>,
            "className": "header__nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Button
                customStyle="card-container__nav--button btn"
                name="Favorites"
/>,
              "className": "",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "customStyle": "card-container__nav--button btn",
                "name": "Favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
